panel/io/state.py

```diff
@@ -333,13 +335,18 @@ async def _scheduled_cb(self, name: str) -> None:
     # Public Methods
     #----------------------------------------------------------------
 
-    def as_cached(self, key: str, fn: Callable[[], None], ttl: int = None, **kwargs) -> None:
+    def as_cached(self, key: str, fn: Callable[[], T], ttl: int = None, **kwargs) -> T:
         """
         Caches the return value of a function, memoizing on the given
         key and supplied keyword arguments.
 
         Note: Keyword arguments must be hashable.
 
+        Example:
+
+        >>> import seaborn as sns
+        >>> planets = pn.state.as_cached("seaborn-planets", sns.load_dataset, name='planets')
```

Comment:
I didn't find the example very intuitive, in particular without knowing what `sns.load_dataset` does and what signature it has. I've come up with this suggestion, feel free to adapt it more if you think it can be improve, or to reject it entirely if you don't like it!

```suggestion
        >>> def load_dataset(name):
        >>>     # some slow operation that uses name to load a dataset....
        >>>     return dataset
        >>> penguins = pn.state.as_cached('dataset-penguins', load_dataset, name='penguins')
```